src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -295,22 +295,43 @@ def window(
         start_index = self.datetime_to_index(start)
         end_index = self.datetime_to_index(end)
 
+        return self._wrapped_buffer_window(
+            self._buffer, start_index, end_index, force_copy
+        )
+
+    @staticmethod
+    def _wrapped_buffer_window(
```

Comment:
Does this have to be a static method? I'd suggest to make it a class method as long it's not used outside the class context.

Reply:
Why do you think so, or what is the advantage of a classmethod? I made it static since it's an isolated algorithmic piece which does not need any internal state of the class. Only the buffer argument would change here. 

The main motivation was testing purpose (actually contained multiple bugs before), i.e. separate wrapping algorithm from index logic (which will become more tricky in follow-ups). But it can also easily be stripped out when needed elsewhere.

Reply:
One example for outside usage I could imagine is the periodic feature extractor, which I assume is using similar code.